### PR TITLE
graph/product: add extended modular product function

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

ISTM this is too open to test in a more specific way.

### CODE REVIEW

File: graph/product/product_ext_example_test.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	p := simple.NewDirectedGraph()
+	product.ModularExt(p, people, pattern, func(a, b graph.Edge) bool {
+		return a.(simple.WeightedEdge).Weight() == b.(simple.WeightedEdge).Weight()
+	})
+
+	// Find the maximal cliques in the undirected induction
+	// of the modular product.
+	mc := topo.BronKerbosch(undirected{p})
+
+	// Report the the cliques that are identical in order to the pattern.
```
Comment (vladimir-ch) [inline graph/product/product_ext_example_test.go]: Double "the".

Comment (kortschak) [reply graph/product/product_ext_example_test.go]: Fixed

**Comment by kortschak** [comment]
I think I'm actually OK with how the testing works. The extended function agrees with the plain modular product with this accept all edges agreement. The function is pretty trivial.

**Comment by vladimir-ch** [comment]
Sure, no problem.

**Pull request merged**